src/widget/wtracktableview.cpp

```diff
@@ -635,6 +635,42 @@ void WTrackTableView::loadSelectionToGroup(QString group, bool play) {
     }
 }
 
+void WTrackTableView::assignPreviousTrackColor() {
+    QModelIndexList indices = selectionModel()->selectedRows();
+    if (indices.size() <= 0) {
+        return;
+    }
+
+    QModelIndex index = indices.at(0);
+    TrackModel* trackModel = getTrackModel();
+    TrackPointer pTrack;
+    if (trackModel &&
+            (pTrack = trackModel->getTrack(index))) {
```

Comment:
Do not do assignment in `if` statements! This is an easy source of bugs. IMO this is an antifeature of C++.

Reply:
Ok, that was just copy-and-pased from another method. I removed here and also from all other methods in the file.